packages/core/components/DirectoryTree/DirectoryTreeNode.tsx

```diff
@@ -49,6 +50,19 @@ export default function DirectoryTreeNode(props: DirectoryTreeNodeProps) {
         return fileSelection.isFocused(fileSet);
     }, [fileSelection, fileSet, collapsed]);
 
+    // Check file count to avoid displaying empty folders
```

Comment:
Hmm why did you decide to go with the route of having each component check asynchronously if it has values as opposed to as the time of gathering the folders to show? This seems like it could cause a lot of network calls.

Reply:
Moved to findChildNodes per offline discussion in [c39b4dc](https://github.com/AllenInstitute/biofile-finder/pull/498/commits/c39b4dc9fc1abc768e6687e4d5c24d6f5a1ed288)